### PR TITLE
Load

### PR DESCRIPTION
**Pull request by AlejandroCano**

### CODE REVIEW

File: Signum.Engine/Engine/SqlPreCommand.cs [hunk 1]
```diff
@@ -180,9 +180,21 @@ public static void ExecuteScript(string title, string script)
...
+                            SafeConsole.WriteLineColor(ConsoleColor.Red, item);
+
+                        }
 
-                    Console.WriteLine();
 
+                    }
+
+                    Console.WriteLine();
+
```
Comment (olmobrutall) [inline Signum.Engine/Engine/SqlPreCommand.cs:197]: Remove spaces

File: Signum.Engine/Engine/Synchronizer.cs [hunk 2]
```diff
@@ -335,7 +338,14 @@ public AutoReplacementContext(string replacementKey, string oldValue, List<strin
...
 
         public static Func<AutoReplacementContext, Selection?>? AutoReplacement;
+        public static Action<string , string , string? >? ResponseRecorder;//  replacementsKey,oldValue,newValue
 
+
+
+
+
+
+        //public static Dictionary<String, Replacements.Selection>? cases ;
```
Comment (olmobrutall) [inline Signum.Engine/Engine/Synchronizer.cs:348]: Remove Spaces

File: Signum.Engine/Engine/Synchronizer.cs [hunk 3]
```diff
@@ -417,8 +446,12 @@ public Selection(string oldValue, string? newValue)
                 this.NewValue = newValue;
             }
 
+            [XmlAttribute]
```
Comment (olmobrutall) [inline Signum.Engine/Engine/Synchronizer.cs:449]: Remove XmlAttribute

File: Signum.Entities/ModifiableEntity.cs [hunk 4]
```diff
@@ -170,6 +170,21 @@ protected virtual void RebindEvents()
                         ((ModifiableEntity)item).SetParentEntity(this);
                 }
             }
+
```
Comment (olmobrutall) [inline Signum.Entities/ModifiableEntity.cs:173]: This is a bad solution, already discussed

File: Signum.React/Scripts/Lines/EntityLine.tsx [hunk 5]
```diff
@@ -121,8 +124,31 @@ export class EntityLineController extends EntityBaseController<EntityLineProps>
 export const EntityLine = React.memo(React.forwardRef(function EntityLine(props: EntityLineProps, ref: React.Ref<EntityLineController>) {
   const c = useController(EntityLineController, props, ref);
   const p = c.props;
+  const [iconString, setIconString] = React.useState<string | null | undefined>(undefined);
```
Comment (olmobrutall) [inline Signum.React/Scripts/Lines/EntityLine.tsx:127]: Use AutocompleteConfig

File: Signum.React/Scripts/Lines/EntityLine.tsx [hunk 6]
```diff
@@ -143,21 +169,32 @@ export const EntityLine = React.memo(React.forwardRef(function EntityLine(props:
...
-              <div className={p.ctx.inputGroupClass}>
-                {renderLink()}
-                {buttons}
-              </div> :
-              renderAutoComplete(input => <div className={p.ctx.inputGroupClass}>
-                {input}
-                {buttons}
-              </div>)
-            )
+      <div className="row">
```
Comment (olmobrutall) [inline Signum.React/Scripts/Lines/EntityLine.tsx:172]: Use AutocompleteConfig

File: Signum.React/Scripts/Lines/EntityListBase.tsx [hunk 7]
```diff
@@ -68,7 +68,7 @@ export abstract class EntityListBaseController<T extends EntityListBaseProps> ex
       return undefined;
 
     return (
-      <a href="#" className={classes("sf-line-button", "sf-move", btn ? "btn btn-light" : undefined)}
+      <a href="#" className={classes("sf-line-button", "sf-move", "sf-move-step", btn ? "btn btn-light" : undefined)}
```
Comment (olmobrutall) [inline Signum.React/Scripts/Lines/EntityListBase.tsx:71]: What is this for?

File: Signum.React/Scripts/Reflection.ts [hunk 8]
```diff
@@ -901,6 +901,16 @@ export function newLite(type: PseudoType, id: number | string): Lite<Entity> {
   };
 }
 
+
+export function newLiteFromKey(key: string): Lite<Entity> {
```
Comment (olmobrutall) [inline Signum.React/Scripts/Reflection.ts:905]: use parseKey

File: Signum.React/Scripts/Lines/EntityRepeater.tsx [hunk 9]
```diff
@@ -14,6 +14,9 @@ export interface EntityRepeaterProps extends EntityListBaseProps {
   createAsLink?: boolean | ((er: EntityRepeaterController) => React.ReactElement<any>);
   avoidFieldSet?: boolean;
   createMessage?: string;
+  getComponentWithIndex?: (ctx: TypeContext<any>, index: number) => React.ReactElement<any>;
```
Comment (olmobrutall) [inline Signum.React/Scripts/Lines/EntityRepeater.tsx:17]: What is this for?

File: Signum.React/Scripts/Lines/EntityListBase.tsx (hunk 7 above)
Comment (AlejandroCano) [reply Signum.React/Scripts/Lines/EntityListBase.tsx:71]: sf-move-step is to be able to add css style (in Lines.css) to the EntityRepeater ordering buttons.

File: Signum.React/Scripts/Lines/EntityRepeater.tsx (hunk 9 above)
Comment (AlejandroCano) [reply Signum.React/Scripts/Lines/EntityRepeater.tsx:17]: this is an alternative to getComponent but the index context is given to the lambda. An example would be if the parent component  should keep a ref to each component

**Pull request merged**